cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java

```diff
@@ -190,6 +177,10 @@ private void doHandleEmbedded(CommandContext ctx, ModelControllerClient client)
             throw new CommandLineException("Failed to execute :reload", e);
         }
 
+        ensureServerRebootComplete(ctx, client);
+    }
+
+    private void ensureServerRebootComplete(CommandContext ctx, ModelControllerClient client) throws CommandLineException {
```

Comment:
Note, the same "making sure it's up" issue affects ShutdownHandler when it's handling a restart.